v2/cli/term/project_page.go

```diff
@@ -0,0 +1,328 @@
+		eventsTable:          tview.NewTable().SetSelectable(true, false),
+		usage:                tview.NewTextView().SetDynamicColors(true),
+	}
+	p.projectInfo.SetBorder(true).SetBorderColor(tcell.ColorWhite)
+	p.eventsTable.SetBorder(true).SetTitle(" Events ")
+	// Create the layout
+	p.page.Flex = tview.NewFlex().
+		SetDirection(tview.FlexRow).
+		AddItem(p.projectInfo, 0, 2, false).
+		AddItem(p.eventsTable, 0, 9, true).
+		AddItem(p.usage, 1, 1, false)
+	return p
+}
+
+func (p *projectPage) load(ctx context.Context, projectID string) {
+	p.refresh(ctx, projectID)
+}
+
+// refresh refreshes Projects info and associated Events and repaints the page.
+func (p *projectPage) refresh(ctx context.Context, projectID string) {
+	if projectID != p.currentProjectID {
+		p.currentProjectID = projectID
+		// "" == continue value for first page
+		p.eventsContinueValues = []string{""}
+	}
+
+	project, err := p.apiClient.Projects().Get(ctx, projectID)
+	if err != nil {
+		// TODO: Handle this
+	}
+	events, err := p.apiClient.Events().List(
+		ctx,
+		&core.EventsSelector{
+			ProjectID: projectID,
+		},
+		&meta.ListOptions{
+			Continue: p.eventsContinueValues[len(p.eventsContinueValues)-1],
+			Limit:    20,
+		},
+	)
+	if err != nil {
+		// TODO: Handle this
+	}
+	p.fillProjectInfo(project)
+	p.fillEventsTable(events)
+	p.fillUsage(events)
+	// Set key handlers
+	p.eventsTable.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		switch event.Key() {
+		case tcell.KeyF5: // Reload
+			p.router.loadProjectPage(projectID)
+		case // Back
+			tcell.KeyLeft,
+			tcell.KeyDelete,
+			tcell.KeyEsc,
+			tcell.KeyBackspace,
+			tcell.KeyBackspace2:
+			p.router.loadProjectsPage()
+		case tcell.KeyRune: // Regular key handling:
+			switch event.Rune() {
+			case 'r', 'R': // Reload
+				p.router.loadProjectPage(projectID)
+			case 'p', 'P': // Previous page
+				// Pop the current page continue value from the stack, but never pop it
+				// if it's the only continue value. i.e. Never pop it if it's the empty
+				// continue value that gets you the first page.
+				if len(p.eventsContinueValues) > 1 {
+					p.eventsContinueValues =
+						p.eventsContinueValues[:len(p.eventsContinueValues)-1]
+					p.router.loadProjectPage(projectID)
+				}
+			case 'n', 'N': // Next page
+				if events.Continue != "" {
+					// Push the continue value for the next page onto the stack
+					p.eventsContinueValues =
+						append(p.eventsContinueValues, events.Continue)
+					p.router.loadProjectPage(projectID)
+				}
+			case 'q', 'Q': // Exit
+				p.router.exit()
+			}
+		}
+		return event
+	})
+
+}
+
+func (p *projectPage) fillProjectInfo(project core.Project) {
+	p.projectInfo.Clear()
+	p.projectInfo.SetTitle(fmt.Sprintf(" %s ", project.ID))
+	infoText := fmt.Sprintf("[grey]Description: [white]%s", project.Description)
+	if project.Spec.WorkerTemplate.Git != nil {
+		infoText = fmt.Sprintf("%s\n[grey]Git:", infoText)
```

Comment:
Do we want to show other fields if non-empty?  (commit, ref, initSubmodules)?

Reply:
Probably. We tried to limit ourselves to the most useful information since every UI puts a premium on screen real estate, but it's admittedly odd to show _some_ git details and not others.